src/main/scala/io/qbeast/spark/delta/IndexFiles.scala

```diff
@@ -89,9 +92,18 @@ object IndexFiles {
       size = indexFile.size,
       modificationTime = indexFile.modificationTime,
       dataChange = dataChange,
+      stats = indexFile.stats,
       tags = tags)
   }
 
+  def fromRemoveFile(removeFile: RemoveFile): IndexFile = {
```

Comment:
I'm not sure it is a good idea to map AddFiles and RemoveFiles to IndexFile. Why would we want to consider IndexFiles, which we use to compute the index status, the RemoveFiles? If we don't want to use the DeleteFiles, I think it is best to create our own hierarchy (e.g., trait QbeastFile, extended by IndexFile and DeletedFile).

Reply:
I created the trait QbeastFile, extended by IndexFile and DeletedFile classes